libs/vkd3d/state.c

```diff
@@ -580,9 +516,15 @@ static HRESULT d3d12_root_signature_init_root_descriptor_tables(struct d3d12_roo
 
         table = &root_signature->parameters[i].u.descriptor_table;
         range_count = p->u.DescriptorTable.NumDescriptorRanges;
+        range_descriptor_offset = 0;
 
         root_signature->parameters[i].parameter_type = p->ParameterType;
         table->range_count = range_count;
+        table->table_index = t++;
+        table->binding_count = 0;
+        table->first_binding = &root_signature->bindings[context->binding_index];
```

Comment:
What is the purpose of first_binding and first_packed_descriptor exactly? I assume this is to hold a range of all CBV/UAV descriptors which don't participate in bindless later?

Reply:
Yes, this is used to iterate over bindings to populate the packed descriptor set.

Reply:
And context->binding_index is currently referring to what? A certain number of descriptors of some other type?

Reply:
It's basically just the current index into the `root_signature->bindings` array. The size is pre-computed, which makes this a bit ugly, but we need to know where to write to.

Reply:
Oh, right, it's updated inside the loop.